main.go

```diff
@@ -0,0 +1,186 @@
+package main
+
+import (
+	"fmt"
+	"os"
+	"strings"
+
+	"github.com/bitrise-io/go-utils/command"
+	"github.com/bitrise-io/go-utils/command/rubycommand"
+	"github.com/bitrise-io/go-utils/errorutil"
+	"github.com/bitrise-io/go-utils/log"
+	"github.com/bitrise-tools/go-steputils/stepconf"
+)
+
+// Config ...
+type Config struct {
+	RepositoryURL string `env:"repository_url,required"`
+
+	GithubAPIToken   stepconf.Secret `env:"github_api_token"`
+	GithubHost       string          `env:"github_host"`
+	GithubAPIBaseURL string          `env:"github_api_base_url"`
+
+	GitlabAPIToken   stepconf.Secret `env:"gitlab_api_token"`
+	GitlabHost       string          `env:"gitlab_host"`
+	GitlabAPIBaseURL string          `env:"gitlab_api_base_url"`
+}
+
+func main() {
+	var cfg Config
+	if err := stepconf.Parse(&cfg); err != nil {
+		failf("Issue with input: %s", err)
+	}
+
+	// Fix repository URL
+	cfg.RepositoryURL = strings.TrimLeft(cfg.RepositoryURL, "https://")
+
+	stepconf.Print(cfg)
+	fmt.Println()
+
+	validateInputs(cfg)
+
+	//
+	// Set local envs for the step
+	{
+		if err := os.Setenv("GIT_REPOSITORY_URL", cfg.RepositoryURL); err != nil {
+			failf("Failed to set env GIT_REPOSITORY_URL, error: %s", err)
+		}
+
+		// Github
+		{
+			if string(cfg.GithubAPIToken) != "" {
```

Comment:
please use an iteration instead, for example:
```
for key, value := range map[string]string{}{
  "DANGER_GITHUB_API_TOKEN":       string(cfg.GithubAPIToken),
  "DANGER_GITHUB_HOST":                string(cfg.GithubHost),
  "DANGER_GITHUB_API_BASE_URL": string(cfg.GithubAPIBaseURL),
  "DANGER_GITLAB_API_TOKEN":        string(cfg.GitlabAPIToken),
  "DANGER_GITLAB_HOST":                  string(cfg.GitlabHost),
  "DANGER_GITLAB_API_BASE_URL":   string(cfg.GitlabAPIBaseURL),
}{
  if value != "" {
	if err := os.Setenv(key, value); err != nil {
		failf("Failed to set env %s, error: %s", key, err)
	}
  }
}
```

(sorry, it is from memory, you might need to fix that)